### DISCUSSION TITLE
Heroku idle mitigation

### DISCUSSION THREAD

**Discussion by anphetamina** [telegraf/telegraf#1416] [Q&A] [answered]
After some time (~45 minutes I guess) a bot deployed on Heroku will exit with status 143 (SIGTERM) after going to idle then restart.

How do you deal with that assuming that webhook is being used?

**Comment by wojpawlik** [comment, marked as answer]
[Store all state in a database](https://12factor.net/processes) or buy a [hobby dyno](https://www.heroku.com/pricing#dynos).
